src/utils/assertions.R

```diff
@@ -0,0 +1,69 @@
+box::use(dplyr)
+box::use(purrr)
+box::use(sf)
+box::use(units)
+
+#' assert_within_distance
+#' Assert that `x` is within `dist` distance from `y`.
+#' @param x sf class with geometry
+#' @param y sf class with geometry
+#' @param dist distance in meters (default = 10000)
+#'
+#' @return TRUE if all elements of x fall within `dist` distance of y.
+#'     Return error if any elements fall outside of distance
+#' @export
+#'
+#' @examples
+#' library(sf)
+#' library(dplyr)
+#' library(units)
+#' file_name <- system.file("shape/nc.shp", package="sf")
+#' nc_counties <- st_read(file_name)
+#' alleghany_county <- nc_counties |>
+#'     filter(NAME == "Alleghany")
+#' # sample points for teseting
+#' set.seed(1)
+#' # these will all be inside the polygon
+#' pts_sampled <- st_sample(
+#'    x = alleghany_county,
+#'    size = 500
+#'    )
+#' # these should have some falling outside
+#' # since we are sampling the bbox some should be outside
+#'  pts_sampled_bbox <- st_sample(
+#'    x = st_bbox(alleghany_county) |>
+#'      st_as_sfc(),
+#'    size = 500
+#'    )
+#'
+#'  assert_within_distance(
+#'    x = pts_sampled,
+#'    y =alleghany_county,
+#'    dist = 1000
+#'    )
+#'  assert_within_distance(
+#'    pts_sampled_bbox,
+#'    y = alleghany_county,
+#'    dist = 1000
+#'    )
+assert_within_distance <- function(x, y, dist = 10000) {
+  m_within_dist <- sf$st_is_within_distance(
```

Comment:
`sf$st_is_within_distance()` fails with polygons, because it merely asserts that boundaries are within a distance.

```r
library(dplyr)
library(sf)
library(units)

file_name <- system.file("shape/nc.shp", package="sf")
clay <- filter(nc_counties, NAME == "Clay")
macon <- filter(nc_counties, NAME == "Macon")

st_is_within_distance(clay, macon, dist = units::set_units(1, "meters"), sparse = FALSE)
```

That returns `TRUE`.

Reply:
ah yes good catch. Forgot a bout polys. Checkout the quick fix here: [f49bdb2](https://github.com/OCHA-DAP/hdx-signals/pull/113/commits/f49bdb26ec2775d0bf63b10f0847a2656db33c8d). Seems to work well